src/Symfony/Component/Form/Tests/Extension/Core/Type/CheckboxTypeTest.php

```diff
@@ -187,4 +187,17 @@ public function testSubmitNull($expected = null, $norm = null, $view = null)
     {
         parent::testSubmitNull(false, false, null);
     }
+
+    public function testSubmitNullUsesDefaultEmptyData($emptyData = 'empty', $expectedData = true)
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, array(
+            'empty_data' => $emptyData,
+        ));
+        $form->submit(null);
+
+        // view data is transformed to the string true value
+        $this->assertSame('1', $form->getViewData());
```

Comment:
This test is a bit confusing, why do we receive `1` here when the `empty_data` is set to `empty`?

Reply:
We can define any true value as empty data as long as the view transformer supports it as input of `reverseTransform`. In any case the will transform back the norm data to view data, so we end up with `'1'` with is the default output of `transform`.

The same happens for any type using a transformer (almost all those overriding the base one).

If we actually define entries in empty data arrays of date types or `RepeatedType` we get the exact same result due to the transformation process.